connectivity/factory/factory.go

```diff
@@ -0,0 +1,248 @@
+	clientEgressL7HTTPPolicyYAML string
+
+	//go:embed manifests/client-egress-l7-http-named-port.yaml
+	clientEgressL7HTTPNamedPortPolicyYAML string
+
+	//go:embed manifests/client-egress-l7-tls.yaml
+	clientEgressL7TLSPolicyYAML string
+
+	//go:embed manifests/client-egress-l7-http-matchheader-secret.yaml
+	clientEgressL7HTTPMatchheaderSecretYAML string
+
+	//go:embed manifests/echo-ingress-from-cidr.yaml
+	echoIngressFromCIDRYAML string
+)
+
+var (
+	clientLabel  = map[string]string{"name": "client"}
+	client2Label = map[string]string{"name": "client2"}
+
+	// Test order is important
+	primaryTests = []factory{
+		noUnexpectedPacketDrops{},
+		noPolicies{},
+		noPoliciesFromOutside{},
+		noPoliciesExtra{},
+		allowAllExceptWorld{},
+		clientIngress{},
+		clientIngressKnp{},
+		allowAllWithMetricsCheck{},
+		allIngressDeny{},
+		allIngressDenyFromOutside{},
+		allIngressDenyKnp{},
+		allEgressDeny{},
+		allEgressDenyKnp{},
+		allEntitiesDeny{},
+		clusterEntity{},
+		clusterEntityMultiCluster{},
+		hostEntityEgress{},
+		hostEntityIngress{},
+		echoIngress{},
+		echoIngressFromOutside{},
+		echoIngressKnp{},
+		clientIngressIcmp{},
+		clientEgress{},
+		clientEgressKnp{},
+		clientEgressExpression{},
+		clientEgressExpressionKnp{},
+		clientWithServiceAccountEgressToEcho{},
+		clientEgressToEchoServiceAccount{},
+		toEntitiesWorld{},
+		toCidrExternal{},
+		toCidrExternalKnp{},
+		fromCidrHostNetns{},
+		echoIngressFromOtherClientDeny{},
+		clientIngressFromOtherClientIcmpDeny{},
+		clientEgressToEchoDeny{},
+		clientIngressToEchoNamedPortDeny{},
+		clientEgressToEchoExpressionDeny{},
+		clientWithServiceAccountEgressToEchoDeny{},
+		clientEgressToEchoServiceAccountDeny{},
+		clientEgressToCidrDeny{},
+		clientEgressToCidrDenyDefault{},
+		health{},
+		northSouthLoadbalancing{},
+		podToPodEncryption{},
+		nodeToNodeEncryption{},
+		egressGateway{},
+		egressGatewayExcludedCidrs{},
+		podToNodeCidrpolicy{},
+		northSouthLoadbalancingWithL7Policy{},
+		echoIngressL7{},
+		echoIngressL7NamedPort{},
+		clientEgressL7Method{},
+		clientEgressL7{},
+		clientEgressL7NamedPort{},
+		clientEgressL7TlsDenyWithoutHeaders{},
+		clientEgressL7TlsHeaders{},
+		clientEgressL7SetHeader{},
+		echoIngressAuthAlwaysFail{},
+		echoIngressMutualAuthSpiffe{},
+		podToIngressService{},
+		podToIngressServiceDenyAll{},
+		podToIngressServiceDenyIngressIdentity{},
+		podToIngressServiceDenyBackendService{},
+		podToIngressServiceAllowIngressIdentity{},
+		outsideToIngressService{},
+		outsideToIngressServiceDenyWorldIdentity{},
+		outsideToIngressServiceDenyCidr{},
+		outsideToIngressServiceDenyAllIngress{},
+		dnsOnly{},
+		toFqdns{},
+		podToControlplaneHost{},
+		podToK8sOnControlplane{},
+		podToControlplaneHostCidr{},
+		podToK8sOnControlplaneCidr{},
+	}
+)
+
+type factory interface {
+	name() string
+	shouldRun(version semver.Version, params check.Parameters) bool
+	build(ct *check.ConnectivityTest, templates map[string]string)
```

Comment:
this is just a personal preference but i prefer the signature being:
```
build(templates map[string]string) *check.Test
```
building a test shouldn't depend on `ConnectivityTest`.

Reply:
Completely agree but unfortunately we already have this dependencies in expectations and scenarios:
- https://github.com/cilium/cilium-cli/blob/main/connectivity/suite.go#L352
- https://github.com/cilium/cilium-cli/blob/main/connectivity/suite.go#L1270
- https://github.com/cilium/cilium-cli/blob/main/connectivity/suite.go#L367

It should be improved but this PR is already large.
So, I would prefer to do it in follow up PRs.

Reply:
ok yeah that's unfortunate. let's do a cleanup later